oracle/config/api.go

```diff
@@ -40,5 +51,13 @@ func (c *APIConfig) ValidateBasic() error {
 		return fmt.Errorf("provider timeout must be greater than 0 and less than the interval")
 	}
 
+	if len(c.URL) == 0 {
```

Comment:
https://pkg.go.dev/net/url#ParseRequestURI

Perphaps we also validate the url is valid beyond being empty?